src/dune/build.mli

```diff
@@ -38,6 +38,9 @@ val all : ('a, 'b) t list -> ('a, 'b list) t
     assume it contains no targets. *)
 val lazy_no_targets : ('a, 'b) t Lazy.t -> ('a, 'b) t
 
+(** Delay a static computation until the arrow is evaluated *)
+val delayed : ('a -> 'b) -> ('a, 'b) t
```

Comment:
Wouldn't it make more sense for this to take `unit` as input?

Also i do wonder if it's simplest to just have `Build.lazy` that will also memoize.

Reply:
Yh, it makes sense for `unit`. Regarding lazy and memoization, we don't require the overhead of memoization here. It's just so that the computation is not forced if the rule is not needed. However, it is effectively evaluated only once.

Reply:
I changed the argument to `unit`